api/internal/plugins/loader/loader.go

```diff
@@ -225,9 +225,26 @@ func (l *Loader) makeBuiltinPlugin(r resid.Gvk) (resmap.Configurable, error) {
 	return nil, errors.Errorf("unable to load builtin %s", r)
 }
 
-func (l *Loader) loadPlugin(res *resource.Resource) (resmap.Configurable, error) {
+func (l *Loader) loadPlugin(ldr ifc.Loader, res *resource.Resource) (resmap.Configurable, error) {
 	spec := fnplugin.GetFunctionSpec(res)
 	if spec != nil {
+		// validation check that function mounts are under the current kustomization directory
+		for _, mount := range spec.Container.StorageMounts {
+			var relPath string
+			var err error
+			if filepath.IsAbs(mount.Src) {
+				relPath, err = filepath.Rel(ldr.Root(), mount.Src)
+				if err != nil {
+					return nil, errors.Wrap(err, "could not convert mount path to relative path")
+				}
+			} else {
+				relPath = mount.Src
+			}
+			if strings.HasPrefix(filepath.Clean(relPath), "../") {
```

Comment:
I agree with your other comment that relative paths should be not only allowed but encouraged, since we want Kustomizations to be encapsulated. Doesn't this still prevent that?

Reply:
> Doesn't this still prevent that?

Could you elaborate a bit more? I'm not sure that I understand what you mean, this particular line will throw an error of the relative path points to something outside of the current directory. 

I did modify this code to disallow absolute paths if that addresses your concerns.